blue_only.py

```diff
@@ -1,221 +1,290 @@
+        0,  # Confirmation
+        33,  # param1: Message ID to be streamed
+        250000,  # param2: Interval in microseconds
+        0,  # param3 (unused)
+        0,  # param4 (unused)
+        0,  # param5 (unused)
+        0,  # param5 (unused)
+        0,  # param6 (unused)
+    )
+
+    secret_key = bytearray(secret_key, "utf-8")
+    vehicle.setup_signing(secret_key, True, my_allow_unsigned_callback, int(time.time()), 0)
+    vehicle.mav.send(pos_message)
+
+    altitude_m = 0
+    loop_counter = 0
+    last_time = current_milli_time()
+    last_image_time = current_milli_time()
+
+    while True:
+        is_pad_detected = False
+        result, image = cam.read()
+        if not result:
+            print("ERROR: Could not get image from camera")
+            continue
+
+        image = cv2.flip(image, 0)
+        image = cv2.flip(image, 1)
+        im_h, im_w, _ = image.shape
+        print("Input image width: " + str(im_w))
+        print("Input image height: " + str(im_h))
+        try:
+            altitude_mm = vehicle.messages[
+                "GLOBAL_POSITION_INT"
+            ].relative_alt  # Note, you can access message fields as attributes!
+            altitude_m = max(altitude_mm / 1000, 0.0)
+            print("Altitude AGL: ", altitude_m)
+        except (KeyError, AttributeError):
+            print("No GLOBAL_POSITION_INT message received")
+            continue
+
+        if args.method == "contour":
+            bounding_box = detect_landing_pads_contour(image)
+        elif args.method == "yolo":
+            bounding_box = detect_landing_pads_yolo(image, config)
+
+        if bounding_box is not None:
+            x, y, w, h = bounding_box
+            rect_image = copy.deepcopy(image)
+            cv2.rectangle(rect_image, (int(x), int(y)), (int(x + w), int(y + h)), (0, 255, 0), 2)
+            x_center = x + w / 2
+            y_center = y + h / 2
+            angle_x = (x_center - im_w / 2) * (FOV_X * (math.pi / 180)) / im_w
+            angle_y = (y_center - im_h / 2) * (FOV_Y * (math.pi / 180)) / im_h
+            cv2.circle(rect_image, (int(x_center), int(y_center)), 2, (0, 0, 255), 2)
+            print("X Angle (rad): ", angle_x)
+            print("Y Angle (rad): ", angle_y)
+            target_dist = calc_target_distance(altitude_m, angle_x, angle_y)
+            is_pad_detected = True
+            vehicle.mav.landing_target_send(
+                0,
+                0,
+                pymavlink.mavutil.mavlink.MAV_FRAME_BODY_NED,
+                angle_x,
+                angle_y,
+                target_dist,
+                0,
+                0,
+            )
+
+        loop_counter += 1
+        if current_milli_time() - last_time > 1000:
+            print("FPS:", loop_counter)
+            loop_counter = 0
+            last_time = current_milli_time()
+        if current_milli_time() - last_image_time > 100:
+            if is_pad_detected:
```

Comment:
same as above, instead of checking this here, have it exit early so we don't have to use a variable here